# 1. Upstream analysis of ATAC seq data

```
Using stem tissue replicate 1 as an example.
```

## 1.1 Map reads to genome

```shell
## make genome index
bwa index -a bwtsw Papaver_somniferum.ref.fa

## align fastq reads to opium genome
bwa mem -v 3 -t 10  /genome_index/Papaver_somniferum.ref.fa  stem_rep1_clean_1.fq.gz  stem_rep1_clean_2.fq.gz -o ATAC_seq_stem_rep1_bwa.sam

## convert sam file to bam file
samtools view -Sb -@ 10  ATAC_seq_stem_rep1_bwa.sam > ATAC_seq_stem_rep1.bam

## sort bam file
samtools sort -@ 10 -O bam  -o ATAC_seq_stem_rep1_sort.bam -T ATAC_seq_stem_rep1_sort.bam.temp ATAC_seq_stem_rep1.bam

## remove dup
sambamba markdup -r -t 10 --hash-table-size=1000000 ATAC_seq_stem_rep1_sort.bam ATAC_seq_stem_rep1_sort_rmdup.bam

## filter low quality mapped reads
samtools view -b -@ 10 -q 10 -F 12 ATAC_seq_stem_rep1_sort_rmdup.bam > ATAC_seq_stem_rep1_rmdup_filter.bam

## sort bam file & make bam index
## these two files for igv visualization

samtools sort -@ 10 -O bam  -o ./bam_filter/ATAC_seq_stem_rep1_rmdup_filter.bam -T ./bam_filter/ATAC_seq_stem_rep1_rmdup_filter.bam.temp ATAC_seq_stem_rep1_rmdup_filter.bam

cd ./bam_filter
samtools index -@ 10 ATAC_seq_stem_rep1_rmdup_filter.bam ATAC_seq_stem_rep1_rmdup_filter.bam.bai
```

## 1.2 Call peak

```shell
## sort -n for call peak
samtools sort -n -@ 10 -O bam  -o ATAC_seq_stem_rep1_rmdup_filter_sortn.bam -T ATAC_seq_stem_rep1_rmdup_filter_sortn.bam.temp ATAC_seq_stem_rep1_rmdup_filter.bam

## call peak
## it takes three reps to combine to call peak
Genrich  -t  ATAC_seq_stem_rep1_rmdup_filter_sortn.bam,ATAC_seq_stem_rep2_rmdup_filter_sortn.bam,ATAC_seq_stem_rep3_rmdup_filter_sortn.bam -o ATAC_seq_{tissue_class}.narrowPeak  -j  -r  -v
```



# 2. PCA

```
## calculate mapped reads in each region of 11 chromosomes for opium poppy
bedtools makewindows -g poppy.genome -w 20000 > poppy.windows
multiBamSummary BED-file  --BED ./chr/poppy.windows --bamfiles ./*.bam  --numberOfProcessors 20 -out ./chr/readCounts.npz --outRawCounts ./chr/readCounts.tab
```

```R
library(FactoMineR)
library(corrplot)
library(factoextra)
atac_seq <- read.csv("D:/001poppy_atac_new_genome/ATAC_peak/atac_peak_cor/readCounts.tab",sep = "\t",stringsAsFactors = F)


atac_seq <- atac_seq[,-1:-3]


colnames(atac_seq) <- c("Capsule_1","Capsule_2","Capsule_3",
                        "Stem_1","Stem_2","Stem_3",
                        "Fine Root_1","Fine Root_2","Fine Root_3",
                        "Leaf_1","Leaf_2","Leaf_3",
                        "Tap Root_1","Tap Root_2","Tap Root_3",
                        "Petal_1","Petal_2","Petal_3")

plot_atac_seq <- as.data.frame(t(atac_seq))
dim(plot_atac_seq)
plot_atac_seq[,108632] <- c("Capsule","Capsule","Capsule",
                            "Stem","Stem","Stem",
                            "Fine Root","Fine Root","Fine Root",
                            "Leaf","Leaf","Leaf",
                            "Tap Root","Tap Root","Tap Root",
                            "Petal","Petal","Petal")


colnames(plot_atac_seq)[108632] <- "Tissue"


atac_seq_pca <- FactoMineR::PCA(t(atac_seq),scale.unit = FALSE, graph = FALSE,ncp=10)

fviz_eig(atac_seq_pca,addlabels = T)
fviz_pca_ind(atac_seq_pca,axes = c(1,2),labelsize =6, 
             habillage = factor(plot_atac_seq$Tissue),
             palette = c("#E41A1C", "#377EB8" ,"#4DAF4A" ,"#984EA3" ,"#FF7F00"  ,"#A65628"),
             invisible='quali'
) +
  ggforce::geom_mark_ellipse(aes(fill = Groups,
                                 color = Groups)) +
  theme(legend.position = 'bottom') +
  coord_equal()


```

# 3. Distance for THSs to TSS

```R
library(ggplot2)
library(ggpointdensity) 
library(viridis)
library("ChIPseeker")
library("GenomicFeatures")



poppy_TxDb <- makeTxDbFromGFF("D:/001poppy_atac_new_genome/Papaver_somniferum.gene.gff3")
setwd("D:/001poppy_atac_new_genome/ATAC_peak/")
files <- list.files(pattern = "*.narrowPeak")
files_list <- as.list(files)
names(files_list) <- c("Capsule","Petal","Stem","Fine_Root","Leaf","Tap_Root")
peakAnnoList <- lapply(files_list, annotatePeak, TxDb=poppy_TxDb,
                       tssRegion=c(-3000, 3000), verbose=FALSE)

plotDistToTSS(peakAnnoList,ylab = "Opening sites (%) (5'->3')",
              title = paste0("Distribution of opening loci relative to TSS"))

```

# 4. Number of promoter-localized THSs per gene in each tissue

```R
library(ggplot2)
library(ggpointdensity) 
library(viridis)
library("ChIPseeker")
library("GenomicFeatures")
options(stringsAsFactors = FALSE)
poppy_TxDb <- makeTxDbFromGFF("D:/001poppy_atac_new_genome/Papaver_somniferum.gene.gff3")
all_peak_result <- rbind()
kb_3_peak_result<- rbind()
for (tissue in c("capsule","petal","stem","leaf","taproot","fineroot")) {
  X = annotatePeak(paste0("ATAC_seq_",tissue,".narrowPeak"),tssRegion = c(-3000,3000),TxDb = poppy_TxDb)
  perk_related_to_gene <- as.data.frame(as.GRanges(X))
  all_peak <- as.data.frame(table(perk_related_to_gene[,19]))
  all_peak <- as.data.frame(table(all_peak[,2]))
  
  all_peak <- as.matrix(all_peak)
  all_peak <- as.data.frame(all_peak)
  all_peak[,1] <- as.numeric(all_peak[,1])
  all_peak[,2] <- as.numeric(all_peak[,2])
  
  all_peak[,2] <- all_peak[,2]/sum(all_peak[,2])
  all_peak[,3] <- tissue
  perk_related_to_gene  <- perk_related_to_gene[abs(perk_related_to_gene[,21]) <= 3000,]
  kb_3_peak <- as.data.frame(table(perk_related_to_gene[,19]))
  kb_3_peak <- as.data.frame(table(kb_3_peak[,2]))
  
  kb_3_peak <- as.matrix(kb_3_peak)
  kb_3_peak <- as.data.frame(kb_3_peak)
  kb_3_peak[,1] <- as.numeric(kb_3_peak[,1])
  kb_3_peak[,2] <- as.numeric(kb_3_peak[,2])
  
  kb_3_peak[,2] <- kb_3_peak[,2]/sum(kb_3_peak[,2])
  kb_3_peak[,3] <- tissue
  all_peak_result <- rbind(all_peak_result,all_peak)
  kb_3_peak_result <- rbind(kb_3_peak_result,kb_3_peak)
  
}

all_peak_result <- all_peak_result[all_peak_result[,1] <= 6,]
kb_3_peak_result <- kb_3_peak_result[kb_3_peak_result[,1] <= 6,]
colnames(kb_3_peak_result) <- c("Number","Frequence","Tissue")
kb_3_peak_result_plot <- kb_3_peak_result
kb_3_peak_result_plot$Number <- factor(kb_3_peak_result_plot$Number)


library(RColorBrewer)
display.brewer.all()
color_name <- brewer.pal(6,"Accent")

library(ggplot2)
ggplot(data = kb_3_peak_result_plot, mapping = aes(
  x = Tissue, fill = Number, y = Frequence
)) + geom_col()+
  scale_fill_manual(values=color_name)

```



# 5. Tissue specific THSs

```
Using stem tissue as an example.
```

```shell
## Intervene was used to intersect different samples to obtain tissue-specific peaks and overlapped peaks
intervene venn  -i ./ATAC_seq* --save-overlaps

cd /ATAC_peak/Intervene_results/sets

sortBed -i 111111_ATAC_seq_capsule_ATAC_seq_fineroot_ATAC_seq_leaf_ATAC_seq_petal_ATAC_seq_stem_ATAC_seq_taproot.bed > all_overlap_sort.bed

bedtools subtract -a ATAC_seq_stem.narrowPeak -b ./all_overlap_sort.bed -A > stem_specific_peak.bed
sortBed -i stem_specific_peak.bed >./stem_specific_peak_sort.bed

```

