# 1. Upstream analysis of ATAC seq data

```
Using stem tissue replicate 1 as an example.
```

## 1.1 Map reads to genome

```shell
## make genome index
bwa index -a bwtsw Papaver_somniferum.ref.fa

## align fastq reads to opium genome
bwa mem -v 3 -t 10  /genome_index/Papaver_somniferum.ref.fa  stem_rep1_clean_1.fq.gz  stem_rep1_clean_2.fq.gz -o ATAC_seq_stem_rep1_bwa.sam

## convert sam file to bam file
samtools view -Sb -@ 10  ATAC_seq_stem_rep1_bwa.sam > ATAC_seq_stem_rep1.bam

## sort bam file
samtools sort -@ 10 -O bam  -o ATAC_seq_stem_rep1_sort.bam -T ATAC_seq_stem_rep1_sort.bam.temp ATAC_seq_stem_rep1.bam

## remove dup
sambamba markdup -r -t 10 --hash-table-size=1000000 ATAC_seq_stem_rep1_sort.bam ATAC_seq_stem_rep1_sort_rmdup.bam

## filter low quality mapped reads
samtools view -b -@ 10 -q 10 -F 12 ATAC_seq_stem_rep1_sort_rmdup.bam > ATAC_seq_stem_rep1_rmdup_filter.bam

## sort bam file & make bam index
## these two files for igv visualization

samtools sort -@ 10 -O bam  -o ./bam_filter/ATAC_seq_stem_rep1_rmdup_filter.bam -T ./bam_filter/ATAC_seq_stem_rep1_rmdup_filter.bam.temp ATAC_seq_stem_rep1_rmdup_filter.bam

cd ./bam_filter
samtools index -@ 10 ATAC_seq_stem_rep1_rmdup_filter.bam ATAC_seq_stem_rep1_rmdup_filter.bam.bai
```

## 1.2 Call peak

```shell
## sort -n for call peak
samtools sort -n -@ 10 -O bam  -o ATAC_seq_stem_rep1_rmdup_filter_sortn.bam -T ATAC_seq_stem_rep1_rmdup_filter_sortn.bam.temp ATAC_seq_stem_rep1_rmdup_filter.bam

## call peak
## it takes three reps to combine to call peak
Genrich  -t  ATAC_seq_stem_rep1_rmdup_filter_sortn.bam,ATAC_seq_stem_rep2_rmdup_filter_sortn.bam,ATAC_seq_stem_rep3_rmdup_filter_sortn.bam -o ATAC_seq_{tissue_class}.narrowPeak  -j  -r  -v
```

