#  Correlation between BIA genes and tfs

```
metabolite <- read.csv("D:/ATAC_peak/BIA/BIA_gene.csv",stringsAsFactors = F)
bia_net <- read.csv("D:/ATAC_peak/BIA/homer_tf/regulatory_network/bia_tf_gene_network_all_infor.csv",
                    stringsAsFactors = F)
bia_net <- bia_net[,c(2,4,6)]
bia_net <- unique(bia_net)


tf <- unique(bia_net[,1])

bia_net[,4] <- 0
rna_seq <- read.csv("D:/rnaseq/tpm_rep.csv",stringsAsFactors = F)
for (i in 1:495) {
  bia_net[i,4] <- cor(as.numeric(rna_seq[rna_seq[,1] == bia_net[i,2],][,2:19]),
                      as.numeric(rna_seq[rna_seq[,1] == bia_net[i,3],][,2:19]))
}

write.csv(bia_net,"D:/ATAC_peak/BIA/homer_tf/regulatory_network/cor.infor.csv")
```

# heatmap

```

gene_select <- read.csv("D:/001poppy_atac_new_genome/ATAC_peak/BIA/homer/gene_name.csv",stringsAsFactors = F)
colnames(gene_select) <- c("poppy_name","symbol_name","rank")
hb6_chip <- read.csv("D:/001poppy_atac_new_genome/ATAC_peak/BIA/homer_tf/regulatory_network/cor.infor_hb6.csv",stringsAsFactors = F)
gene_select <- gene_select[gene_select[,1] %in% hb6_chip[,2],]


gene_select[20,1] <- "Pso06G40150.0"
gene_select[20,2] <- "HB6"
gene_select[20,3] <- 0
gene_select <- gene_select[order(gene_select$rank),]

gene_select <- gene_select[-c(19,20),]

rna_seq_tpm_normal <- read.csv("D:/001poppy_atac_new_genome/rnaseq/tpm.csv",stringsAsFactors = F,header = T)
colnames(rna_seq_tpm_normal) <- c("gene","Capsule","Petal","Stem","Fine Root","Leaf","Tap Root")

rna_seq <- merge(rna_seq_tpm_normal,gene_select,by.x="gene",by.y="poppy_name")
rna_seq <- rna_seq[order(rna_seq$rank),]
rna_seq[,1] <- paste0(rna_seq[,1],"_",rna_seq[,8])

row.names(rna_seq) <- rna_seq[,1]
rna_seq <- rna_seq[,c(-1,-8,-9)]



library(pheatmap)
pheatmap(as.matrix(rna_seq),scale = "row",cluster_row = FALSE,cluster_cols = TRUE)
pheatmap(log2(as.matrix(rna_seq)+1),scale = "row",cluster_row = FALSE,cluster_cols = TRUE)


```

