```
Note: "tpm_rep.csv" is the gene expression file for six distinctive tissues (each with three replicates) of opium poppy.
```



# 1. Upstream analysis of RNA seq data (conducted by Mengyao Guo)

```shell
#poppy_RNAseq
#-----------------------------------------------------------------
#pathway for files
#pathway for task.pbs
/home/myguo/poppy_RNAseq/pipeline_for_poppy/task.pbs
#pathway for poppy RNA-seq raw data
/home/DATA/Arabidopsis_poppy/X101SC19041787-Z02_yingsu_data_Release_20191217/RNA/yingsu/
#pathway for poppy atac-seq raw data
/home/DATA/Arabidopsis_poppy/poppy_ATAC/xy_merge_all_atac_fq_document_200820
#pathway for reference
/home/xfyang/poppy_project/HN1_DCW_YMR_fianl_uploaded_210510/Papaver_somniferum.*

#pathway for poppy RNA-seq result
/home/myguo/poppy_RNAseq/result

#running 
pbs+sh
#-----------------------------------------------------------------
#software load
conda install -c bioconda fastqc
conda install -c bioconda multiqc
#-----------------------------------------------------------------
#cheek downlod files
#md5sum -c /home/myguo/timetrans/result/2.cleandata/*r/MD5*.txt
#-----------------------------------------------------------------

#-----------------------------------------------------------------
#fastqc
cd /home/DATA/Arabidopsis_poppy/X101SC19041787-Z02_yingsu_data_Release_20191217/RNA/yingsu
for dir in /home/DATA/Arabidopsis_poppy/X101SC19041787-Z02_yingsu_data_Release_20191217/RNA/yingsu/* ; do
	    cd $dir;
	    cd clean ;
        fastqc *.gz -t 28 -o /home/myguo/poppy_RNAseq/result/fastqc;
        cd ..;
        cd ..;
done
echo "OK"
#multiqc
multiqc /home/myguo/poppy_RNAseq/result/fastqc -o /home/myguo/poppy_RNAseq/result/multiqc
#-----------------------------------------------------------------
#bowtie-cufflinks
#-----------------------------------------------------------------
#bowtie2-build index
bowtie2-build /home/xfyang/poppy_project/HN1_DCW_YMR_fianl_uploaded_210510/Papaver_somniferum.ref.fa /home/myguo/poppy_RNAseq/ref/Papaver_somniferum_ref_index
#bowtie2 alignment
#for dir in /home/DATA/Arabidopsis_poppy/X101SC19041787-Z02_yingsu_data_Release_20191217/RNA/yingsu/* ;do echo $dir;done3
cd /home/DATA/Arabidopsis_poppy/X101SC19041787-Z02_yingsu_data_Release_20191217/RNA/yingsu/
for dir in $(ls);do
	echo $dir;
	mkdir /home/myguo/poppy_RNAseq/result/align/$dir/;
	bowtie2 -p 28 -x /home/myguo/poppy_RNAseq/ref/Papaver_somniferum_ref_index -1 /home/DATA/Arabidopsis_poppy/X101SC19041787-Z02_yingsu_data_Release_20191217/RNA/yingsu/$dir/clean/*1_1_clean* -2 /home/DATA/Arabidopsis_poppy/X101SC19041787-Z02_yingsu_data_Release_20191217/RNA/yingsu/$dir/clean/*1_2_clean* -S /home/myguo/poppy_RNAseq/result/align/$dir/$dir.sam &>> /home/myguo/poppy_RNAseq/result/bowtie2alignsum.txt
	echo "#-----------------------------------" >> /home/myguo/poppy_RNAseq/result/bowtie2alignsum.txt
	samtools view -@ 28 -bS -q 10 /home/myguo/poppy_RNAseq/result/align/$dir/$dir.sam > /home/myguo/poppy_RNAseq/result/align/$dir/$dir.bam
	samtools sort -@ 28 /home/myguo/poppy_RNAseq/result/align/$dir/$dir.bam -o /home/myguo/poppy_RNAseq/result/align/$dir/sorted_$dir.bam
	samtools index -@ 28 /home/myguo/poppy_RNAseq/result/align/$dir/sorted_$dir.bam /home/myguo/poppy_RNAseq/result/align/$dir/sorted_$dir.bam.index
	rm /home/myguo/poppy_RNAseq/result/align/$dir/$dir.sam
	rm /home/myguo/poppy_RNAseq/result/align/$dir/$dir.bam
	samtools flagstat -@ 28 /home/myguo/poppy_RNAseq/result/align/$dir/*.bam >> /home/myguo/poppy_RNAseq/result/flagstat.txt
	echo "#-----------------------------------" >> /home/myguo/poppy_RNAseq/result/flagstat.txt
	#bam_stat.py -i /home/myguo/timetrans/result/2.cleandata/$dir/sorted_$dir.bam > /home/myguo/timetrans/result/2.cleandata/$dir/bam_stat_$dir.log
	cd ..;
done
#------------------------------------------------------------------
#cufflinks-cuffmerge-cuffdiff

source activate python27
#gff3 to gtf
gffread /home/xfyang/poppy_project/HN1_DCW_YMR_fianl_uploaded_210510/Papaver_somniferum.gene.gff3 -T -o /home/myguo/poppy_RNAseq/ref/ref/Papaver_somniferum.gene.gtf
#cufflinks
cd /home/myguo/poppy_RNAseq/result/align/
for dir in $(ls);do
	echo $dir;
	cd $dir;
    mkdir /home/myguo/poppy_RNAseq/result/align/$dir/cufflinksG/;
    cufflinks -o /home/myguo/poppy_RNAseq/result/align/$dir/cufflinksG/cufflink_$dir -p 28 -G /home/myguo/poppy_RNAseq/ref/ref/Papaver_somniferum.gene.gtf /home/myguo/poppy_RNAseq/result/align/$dir/*.bam
    cd ..;
done
echo "OK" 

cd /home/myguo/poppy_RNAseq/result/align/
for dir in $(ls); do
	cd /home/myguo/poppy_RNAseq/result/align/$dir/cufflinksG/cufflink_$dir/;
    for file in /home/myguo/poppy_RNAseq/result/align/$dir/cufflinksG/cufflink_$dir/* ;do  
    	echo $file >> /home/myguo/poppy_RNAseq/result/assembly_for_cuff_G.txt; done
    cd /home/myguo/poppy_RNAseq/result/align/;
done
echo "OK"

awk 'NR%4'==0 /home/myguo/poppy_RNAseq/result/assembly_for_cuff_G.txt > /home/myguo/poppy_RNAseq/result/assembly_trans_cuff_G.txt
#cuffmerge -o /home/myguo/poppy_RNAseq/result/merged_asm_G -g /home/myguo/poppy_RNAseq/ref/ref/Papaver_somniferum.gene.gtf -s /home/xfyang/poppy_project/HN1_DCW_YMR_fianl_uploaded_210510/Papaver_somniferum.ref.fa -p 28 /home/myguo/poppy_RNAseq/result/assembly_trans_cuff_G.txt
echo "OK"
#cuffdiff
#awk 'NR%4'==3 /home/myguo/poppy_RNAseq/result/sorted_bam_full_pathway.txt > /home/myguo/poppy_RNAseq/result/sorted_bam_pathway.txt.txt
cuffdiff -o /home/myguo/poppy_RNAseq/result/diff_tissue_A_out --labels Capsule,Petal,Stem,Fineroot,Leaf,Taproot --library-norm-method classic-fpkm -b /home/xfyang/poppy_project/HN1_DCW_YMR_fianl_uploaded_210510/Papaver_somniferum.ref.fa -p 27 -u /home/myguo/poppy_RNAseq/ref/ref/Papaver_somniferum.gene.gtf /home/myguo/poppy_RNAseq/result/align/guo/sorted_guo_rep1.bam,/home/myguo/poppy_RNAseq/result/align/guo2/sorted_guo_rep2.bam,/home/myguo/poppy_RNAseq/result/align/guo3/sorted_guo_rep3.bam /home/myguo/poppy_RNAseq/result/align/hua/sorted_hua_rep1.bam,/home/myguo/poppy_RNAseq/result/align/hua2/sorted_hua_rep2.bam,/home/myguo/poppy_RNAseq/result/align/hua3/sorted_hua_rep3.bam /home/myguo/poppy_RNAseq/result/align/jing/sorted_jing_rep1.bam,/home/myguo/poppy_RNAseq/result/align/jing2/sorted_jing_rep2.bam,/home/myguo/poppy_RNAseq/result/align/jing3/sorted_jing_rep3.bam /home/myguo/poppy_RNAseq/result/align/xugen1/sorted_xugen_rep1.bam,/home/myguo/poppy_RNAseq/result/align/xugen2/sorted_xugen_rep2.bam,/home/myguo/poppy_RNAseq/result/align/xugen3/sorted_xugen_rep3.bam /home/myguo/poppy_RNAseq/result/align/ye/sorted_ye_rep1.bam,/home/myguo/poppy_RNAseq/result/align/ye2/sorted_ye_rep2.bam,/home/myguo/poppy_RNAseq/result/align/ye3/sorted_ye_rep3.bam /home/myguo/poppy_RNAseq/result/align/zhugen1/sorted_zhugen_rep1.bam,/home/myguo/poppy_RNAseq/result/align/zhugen2/sorted_zhugen_rep2.bam,/home/myguo/poppy_RNAseq/result/align/zhugen3/sorted_zhugen_rep3.bam
#---------------------------------------------------------------------
#---------------------------------------------------------------------
#FPKM to TPM
#---------------------------------------------------------------------

external_scaled_frags<-read.table("genes.read_group_tracking",header=TRUE,sep="\t")
sample<-c("Capsule","Petal","Stem","Fineroot","Leaf","Taproot")
sample_length<-c(1:length(sample))
replicate<-c("0","1","2")
select_gene_read_group_tracking<-data.frame()
for(i in sample_length){
  for(j in c(1:length(replicate))){
    single_sample<-subset(external_scaled_frags,condition==sample[i] & replicate==replicate[j],select=c(tracking_id,external_scaled_frags,FPKM))
    names(single_sample)<-c(paste(sample[i],replicate[j],"tracking_id",sep="_"),paste(sample[i],replicate[j],sep=""),paste(sample[i],replicate[j],sep=""))
    assign(paste(sample[i],replicate[j],sep=""),single_sample)
}
}
select_gene_read_group_tracking<-cbind( Capsule0,Capsule1,Capsule2,Petal0,Petal1,Petal2,Stem0,Stem1,Stem2,Fineroot0,Fineroot1,Fineroot2,Leaf0,Leaf1,Leaf2,Taproot0,Taproot1,Taproot2 )
select_external_frags<-cbind(select_gene_read_group_tracking[,1],select_gene_read_group_tracking[,seq(2,ncol(select_gene_read_group_tracking),3)])
select_external_frags_fpkm<-cbind(select_gene_read_group_tracking[,1],select_gene_read_group_tracking[,seq(3,ncol(select_gene_read_group_tracking),3)])
#change colnames1 to tracking_id
colnames(select_external_frags)[1]<-'tracking_id'
colnames(select_external_frags_fpkm)[1]<-'tracking_id'
#SET TRACKIG-ID AS ROWNAMES
rownames(select_external_frags_fpkm)<-select_external_frags_fpkm[,1]
select_external_frags_fpkm<-select_external_frags_fpkm[,-1]
rownames(select_external_frags)<-select_external_frags[,1]
select_external_frags<-select_external_frags[,-1]
fpkmToTpm <- function(fpkm)
{
    exp(log(fpkm) - log(sum(fpkm)) + log(1e6))
}
tpms <- apply(select_external_frags_fpkm,2,fpkmToTpm)
colSums(tpms)
nn<-c( "Capsule1","Capsule0","Capsule2","Petal1","Petal0","Petal2","Stem1","Stem0","Stem2","Fineroot1","Fineroot0","Fineroot2","Leaf1","Leaf0","Leaf2","Taproot1","Taproot0","Taproot2" )
> nn
 [1] "Capsule1"  "Capsule0"  "Capsule2"  "Petal1"    "Petal0"   
 [6] "Petal2"    "Stem1"     "Stem0"     "Stem2"     "Fineroot1"
[11] "Fineroot0" "Fineroot2" "Leaf1"     "Leaf0"     "Leaf2"    
[16] "Taproot1"  "Taproot0"  "Taproot2" 
colnames(select_external_frags)<-nn
colnames(select_external_frags_fpkm)<-nn
colnames(tpms)<-nn
write.csv(select_external_frags,file="select_external_frags.csv")
write.csv(select_external_frags_fpkm,file="select_external_frags_fpkm.csv")
write.csv(tpms,file="select_external_frags_tpm.csv")

```



# 2. PCA

```R
install.packages("FactoMineR")
install.packages("factoextra")
library(FactoMineR)
library(corrplot)
library(factoextra)
rna_seq <- read.csv("D:/001poppy_atac_new_genome/rnaseq/tpm_rep.csv",stringsAsFactors = F)

row.names(rna_seq) <- rna_seq[,1]
rna_seq <- rna_seq[,-1]

colnames(rna_seq)
colnames(rna_seq) <- c("Capsule_1","Capsule_2","Capsule_3",
                       "Petal_1","Petal_2","Petal_3",
                       "Stem_1","Stem_2","Stem_3",
                       "Fine Root_1","Fine Root_2","Fine Root_3",
                       "Leaf_1","Leaf_2","Leaf_3",
                       "Tap Root_1","Tap Root_2","Tap Root_3")

plot_rna_seq <- as.data.frame(t(rna_seq))
dim(plot_rna_seq)
plot_rna_seq[,51217] <- c("Capsule","Capsule","Capsule",
                          "Petal","Petal","Petal",
                          "Stem","Stem","Stem",
                          "Fine Root","Fine Root","Fine Root",
                          "Leaf","Leaf","Leaf",
                          "Tap Root","Tap Root","Tap Root")


colnames(plot_rna_seq)[51217] <- "Tissue"

rna_seq_pca <- FactoMineR::PCA(t(rna_seq),graph = FALSE)
fviz_eig(rna_seq_pca,addlabels = T)

library(RColorBrewer)
library(ggforce)
brewer.pal(9,"Set1")
fviz_pca_ind(rna_seq_pca,axes = c(1,2),labelsize =6, 
             habillage = factor(plot_rna_seq$Tissue),
             palette = c("#E41A1C", "#377EB8" ,"#4DAF4A" ,"#984EA3" ,"#FF7F00"  ,"#A65628"),
) +
  ggforce::geom_mark_ellipse(aes(fill = Groups,
                                 color = Groups)) +
  theme(legend.position = 'bottom') +
  coord_equal()

```



# 3. Venn

```R
rna_seq_tpm_normal <- read.csv("tpm.csv",stringsAsFactors = F,header = T)
colnames(rna_seq_tpm_normal) <- c("gene","capsule","petal","stem","finroot","leaf","taproot")

x <- list(guo = rna_seq_tpm_normal[rna_seq_tpm_normal[,2]>=1,][,1], hua = rna_seq_tpm_normal[rna_seq_tpm_normal[,3]>=1,][,1] , jing = rna_seq_tpm_normal[rna_seq_tpm_normal[,4]>=1,][,1], xugen = rna_seq_tpm_normal[rna_seq_tpm_normal[,5]>=1,][,1], ye = rna_seq_tpm_normal[rna_seq_tpm_normal[,6]>=1,][,1], zhugen = rna_seq_tpm_normal[rna_seq_tpm_normal[,7]>=1,][,1])

library(venn)
venn(x,opacity = 0.4, plotsize = 20, ilcs = 2, sncs = 1.5,zcolor = "style")

```



# 4. Co-expressed gene modules identified

```R

#install.packages("BiocManager")
library("BiocManager")
#BiocManager::install("WGCNA")
#BiocManager::install("gplots")
library("WGCNA")
library("gplots")


##1 Data input, cleaning and pre-processing
##1.a Loading expression data
getwd()

setwd("/r_script/wgcna_poppy_atac/")
femData = read.csv("tpm_rep.csv",stringsAsFactors = F)
## Note that each row corresponds to a gene and column to a sample or auxiliary information.
## We now remove the auxiliary data and transpose the expression data for further analysis.
row.names(femData) <- femData[,1]
femData <- femData[,-1]

## filtering out non-expressed genes

m.var = apply(femData,1,sum)
femData = femData[which(m.var>3),]
datExpr0 = as.data.frame(t(femData))

##1.b Checking data for excessive missing values and identification of outlier microarray samples
## We first check for genes and samples with too many missing values
gsg = goodSamplesGenes(datExpr0, verbose = 3);
gsg$allOK

##If the last statement returns TRUE, all genes have passed the cuts. If not, we remove the offending genes and samples from the data:
if (!gsg$allOK){
  # Optionally, print the gene and sample names that were removed:
  if (sum(!gsg$goodGenes)>0)
    printFlush(paste("Removing genes:", paste(names(datExpr0)[!gsg$goodGenes], collapse = ", ")));
  if (sum(!gsg$goodSamples)>0)
    printFlush(paste("Removing samples:", paste(rownames(datExpr0)[!gsg$goodSamples], collapse = ", ")));
  # Remove the offending genes and samples from the data:
  datExpr0 = datExpr0[gsg$goodSamples, gsg$goodGenes]
}

## Next we cluster the samples (in contrast to clustering genes that will come later) to see if there are any obvious outliers.

sampleTree = hclust(dist(datExpr0),method = "complete");
# Plot the sample tree: Open a graphic output window of size 12 by 9 inches
# The user should change the dimensions if the window is too large or too small.
##sizeGrWindow(12,9)
pdf(file = "001_sampleClustering.pdf", width = 12, height = 9);
par(cex = 0.6);
par(mar = c(0,4,2,0))
plot(sampleTree, main = "Sample clustering to detect outliers", sub="", xlab="", cex.lab = 1.5,
     cex.axis = 1.5, cex.main = 2)
dev.off()


datExpr = datExpr0
nGenes = ncol(datExpr)
nSamples = nrow(datExpr)
##1.c Loading clinical trait data
traitData = read.csv("trait.csv")
rownames(traitData) = traitData[,1]
traitData <- traitData[,-1]
datTraits = traitData

##We now have the expression data in the variable datExpr, and the corresponding clinical traits in the variable datTraits.
##Before we continue with network construction and module detection, we visualize how the clinical traits relate to the sample dendrogram.
# Re-cluster samples
sampleTree2 = hclust(dist(datExpr), method = "complete")
# Convert traits to a color representation: white means low, red means high, grey means missing entry
traitColors = numbers2colors(datTraits, signed = FALSE);
# Plot the sample dendrogram and the colors underneath.
pdf(file = "002_sampleClustering_with_trait.pdf", width = 12, height = 9);
plotDendroAndColors(sampleTree2, traitColors,
                    groupLabels = names(datTraits),
                    main = "Sample dendrogram and trait heatmap")
dev.off()
save(datExpr, datTraits, file = "01-dataInput.RData")

##2 Automatic network construction and module detection

rm(list=ls())
enableWGCNAThreads()
lnames = load(file = "01-dataInput.RData");
#The variable lnames contains the names of loaded variables.
lnames
# Choose a set of soft-thresholding powers
powers = c(c(1:10), seq(from = 12, to=20, by=2))
# Call the network topology analysis function
sft = pickSoftThreshold(datExpr, powerVector = powers, verbose = 5)
pdf(file = "003_soft-thresholding_power.pdf");
# Plot the results:
par(mfrow = c(1,2));
cex1 = 0.9;
# Scale-free topology fit index as a function of the soft-thresholding power
plot(sft$fitIndices[,1], -sign(sft$fitIndices[,3])*sft$fitIndices[,2],
     xlab="Soft Threshold (power)",ylab="Scale Free Topology Model Fit,signed R^2",type="n",
     main = paste("Scale independence"));
text(sft$fitIndices[,1], -sign(sft$fitIndices[,3])*sft$fitIndices[,2],
     labels=powers,cex=cex1,col="red");
# this line corresponds to using an R^2 cut-off of h
abline(h=0.90,col="red")
# Mean connectivity as a function of the soft-thresholding power
plot(sft$fitIndices[,1], sft$fitIndices[,5],
     xlab="Soft Threshold (power)",ylab="Mean Connectivity", type="n",
     main = paste("Mean connectivity"))
text(sft$fitIndices[,1], sft$fitIndices[,5], labels=powers, cex=cex1,col="red")
dev.off()
## net$colors contains the module assignment, and net$MEs contains the module eigengenes of the modules.
sft$powerEstimate
##Co-expression similarity and adjacency
softPower = sft$powerEstimate
adjacency = adjacency(datExpr, power = softPower);
##Topological Overlap Matrix (TOM)
##To minimize effects of noise and spurious associations, we transform the adjacency into Topological Overlap Matrix,
##and calculate the corresponding dissimilarity:
# Turn adjacency into topological overlap
TOM = TOMsimilarity(adjacency);
dissTOM = 1-TOM

##Clustering using TOM
##We now use hierarchical clustering to produce a hierarchical clustering tree (dendrogram) of genes. Note that we use
##the function hclust that provides a much faster hierarchical clustering routine than the standard hclust function.
# Call the hierarchical clustering function
geneTree = hclust(as.dist(dissTOM), method = "complete");
# Plot the resulting clustering tree (dendrogram)
pdf(file = "004_Gene_clustering_on_TOM-based_dissimilarity.pdf");
##sizeGrWindow(12,9)
plot(geneTree, xlab="", sub="", main = "Gene clustering on TOM-based dissimilarity",
     labels = FALSE, hang = 0.04);
dev.off()

png(file = "004_Gene_clustering_on_TOM-based_dissimilarity.png");
##sizeGrWindow(12,9)
plot(geneTree, xlab="", sub="", main = "Gene clustering on TOM-based dissimilarity",
     labels = FALSE, hang = 0.04);
dev.off()

# We like large modules, so we set the minimum module size relatively high:
minModuleSize = 100
# Module identification using dynamic tree cut:
dynamicMods = cutreeDynamic(dendro = geneTree, distM = dissTOM,
                            deepSplit = 2, pamRespectsDendro = FALSE,
                            minClusterSize = minModuleSize);
##Label 0 is reserved for unassigned genes.
table(dynamicMods)
##We now plot the module assignment under the gene dendrogram
# Convert numeric lables into colors
dynamicColors = labels2colors(dynamicMods)
table(dynamicColors)
# Plot the dendrogram and colors underneath
pdf(file = "005_dendrogram_and_colors_underneath.pdf");
##sizeGrWindow(8,6)
plotDendroAndColors(geneTree, dynamicColors, "Dynamic Tree Cut",
                    dendroLabels = FALSE, hang = 0.03,
                    addGuide = TRUE, guideHang = 0.05,
                    main = "Gene dendrogram and module colors")
dev.off()
##Merging of modules whose expression profiles are very similar
##The Dynamic Tree Cut may identify modules whose expression profiles are very similar. It may be prudent to merge
##such modules since their genes are highly co-expressed. To quantify co-expression similarity of entire modules, we
##calculate their eigengenes and cluster them on their correlation:
# Calculate eigengenes
MEList = moduleEigengenes(datExpr, colors = dynamicColors)
MEs = MEList$eigengenes
# Calculate dissimilarity of module eigengenes
MEDiss = 1-cor(MEs);
# Cluster module eigengenes
METree = hclust(as.dist(MEDiss), method = "complete");
# Plot the result
pdf(file = "006_Clustering_of_module_eigengenes.pdf");
##sizeGrWindow(7, 6)
plot(METree, main = "Clustering of module eigengenes",
     xlab = "", sub = "")

MEDissThres = 0.25
# Plot the cut line into the dendrogram
abline(h=MEDissThres, col = "red")
dev.off()
# Call an automatic merging function
merge = mergeCloseModules(datExpr, dynamicColors, cutHeight = MEDissThres, verbose = 3)
# The merged module colors
mergedColors = merge$colors;
# Eigengenes of the new merged modules:
mergedMEs = merge$newMEs;
pdf(file = "007_geneDendro-3.pdf");
##sizeGrWindow(12, 9)
#pdf(file = "Plots/geneDendro-3.pdf", wi = 9, he = 6)
plotDendroAndColors(geneTree, cbind(dynamicColors, mergedColors),
                    c("Dynamic Tree Cut", "Merged dynamic"),
                    dendroLabels = FALSE, hang = 0.03,
                    addGuide = TRUE, guideHang = 0.05)
dev.off()
##In the subsequent analysis, we will use the merged module colors in mergedColors. We save the relevant variables for
##use in subsequent parts of the tutorial:
# Rename to moduleColors
moduleColors = mergedColors
# Construct numerical labels corresponding to the colors
colorOrder = c("grey", standardColors(50));
moduleLabels = match(moduleColors, colorOrder)-1;
MEs = mergedMEs;
# Save module colors and labels for use in subsequent parts
save(MEs, moduleLabels, moduleColors, geneTree, file = "02-networkConstruction-auto.RData")

##3. Relating modules to external information and identifying important genes
rm(list=ls())
lnames = load(file = "01-dataInput.RData");
lnames = load(file = "02-networkConstruction-auto.RData");

# Define numbers of genes and samples
nGenes = ncol(datExpr);
nSamples = nrow(datExpr);
# Recalculate MEs with color labels
MEs0 = moduleEigengenes(datExpr, moduleColors)$eigengenes
MEs = orderMEs(MEs0)
moduleTraitCor = cor(MEs, datTraits, use = "p");
moduleTraitPvalue = corPvalueStudent(moduleTraitCor, nSamples);

# Will display correlations and their p-values
textMatrix = paste(signif(moduleTraitCor, 2), "\n(",
                   signif(moduleTraitPvalue, 1), ")", sep = "");

dim(textMatrix) = dim(moduleTraitCor)
pdf(file = "008_labeledHeatmap.pdf");
##sizeGrWindow(10,6)
par(mar = c(6, 8.5, 3, 3));
# Display the correlation values within a heatmap plot
labeledHeatmap(Matrix = moduleTraitCor,
               xLabels = names(datTraits),
               yLabels = names(MEs),
               ySymbols = names(MEs),
               colorLabels = FALSE,
               colors = greenWhiteRed(50),
               textMatrix = textMatrix,
               setStdMargins = FALSE,
               cex.text = 0.5,
               zlim = c(-1,1),
               main = paste("Module-trait relationships"))

dev.off()
##Output gene lists
intModules = as.character(as.data.frame(table(moduleColors))[,1])
probes = names(datExpr)

for (module in intModules){
  # Select module probes
  modGenes = (moduleColors==module)
  # Get their entrez ID codes
  modLLIDs = probes[modGenes];
  # Write them into a file
  fileName = paste0( module, "_gene_name.csv", sep="");
  write.csv(as.data.frame(modLLIDs), file = fileName,
            row.names = FALSE)
}
rm(list = ls())
options(stringsAsFactors = FALSE);
lnames = load(file = "01-dataInput.RData");
lnames = load(file = "02-networkConstruction-auto.RData");
nGenes = ncol(datExpr)
nSamples = nrow(datExpr)

##Visualizing the gene network
# Calculate topological overlap anew: this could be done more efficiently by saving the TOM
# calculated during module detection, but let us do it again here.
dissTOM = 1-TOMsimilarityFromExpr(datExpr, power = 14);
# Transform dissTOM with a power to make moderately strong connections more visible in the heatmap
plotTOM = dissTOM^7;
# Set diagonal to NA for a nicer plot
diag(plotTOM) = NA;
# Call the plot function

##pdf(file = "009_tom_plot.pdf", wi = 9, he = 9)
##sizeGrWindow(9,9)
##myheatcol = colorpanel(250,'red',"orange",'lemonchiffon')
##TOMplot(plotTOM, geneTree, moduleColors, main = "Network heatmap plot, all genes",col=myheatcol)
##dev.off()


##Note that the generating the heatmap plot may take a substantial amount of time. It is possible to restrict the
##number of genes to speed up the plotting; however, the gene dendrogram of a subset of genes will often look different
##from the gene dendrogram of all genes. In the following example we restrict the number of plotted genes to 400:
nSelect = 5000
# For reproducibility, we set the random seed
iset.seed(20210613);
select = sample(nGenes, size = nSelect);
selectTOM = dissTOM[select, select];
# There’s no simple way of restricting a clustering tree to a subset of genes, so we must re-cluster.
selectTree = hclust(as.dist(selectTOM), method = "complete")
selectColors = moduleColors[select];
# Open a graphical window
sizeGrWindow(9,9)
# Taking the dissimilarity to a power, say 10, makes the plot more informative by effectively changing
# the color palette; setting the diagonal to NA also improves the clarity of the plot
plotDiss = selectTOM^7;
diag(plotDiss) = NA;
pdf(file = "009_tom_plot_select.pdf", wi = 9, he = 9)
##sizeGrWindow(9,9)
myheatcol = colorpanel(250,'red',"orange",'lemonchiffon')
TOMplot(plotDiss, selectTree, selectColors, main = "Network heatmap plot, selected genes")
dev.off()

png(file = "009_tom_plot_select.png", wi = 9, he = 9)
##sizeGrWindow(9,9)
myheatcol = colorpanel(250,'red',"orange",'lemonchiffon')
TOMplot(plotDiss, selectTree, selectColors, main = "Network heatmap plot, selected genes")
dev.off()

options(stringsAsFactors = FALSE);
rm(list=ls())
lnames = load(file = "01-dataInput.RData");
lnames = load(file = "02-networkConstruction-auto.RData");
# Recalculate topological overlap if needed
TOM = TOMsimilarityFromExpr(datExpr, power = 14);
# Select modules
modules_all = as.character(as.data.frame(table(moduleColors))[,1])

for(modules in modules_all ){
  probes = names(datExpr)
  inModule = is.finite(match(moduleColors, modules));
  modProbes = probes[inModule];
  # Select the corresponding Topological Overlap
  modTOM = TOM[inModule, inModule]
  dimnames(modTOM) = list(modProbes, modProbes)
  # Export the network into edge and node list files Cytoscape can read
  cyt = exportNetworkToCytoscape(modTOM,
                                 edgeFile = paste("CytoscapeInput-edges-", paste(modules, collapse="-"), ".txt", sep=""),
                                 nodeFile = paste("CytoscapeInput-nodes-", paste(modules, collapse="-"), ".txt", sep=""),
                                 weighted = TRUE,
                                 threshold = 0.02,
                                 nodeNames = modProbes,
                                 nodeAttr = moduleColors[inModule])
  
}


```



