

# 1. Joint analysis of RNA-seq and ATAC-seq identifies tissue-specific hub transcription factors

```
Using stem tissue as an example.
```



## 1.1 assign THSs to module genes

```R
library(ggplot2)
library(ggpointdensity) 
library(viridis)
library("ChIPseeker")
library("GenomicFeatures")

tissue <- "stem"
module <- "blue"

poppy_TxDb <- makeTxDbFromGFF("Papaver_somniferum.gene.gff3")

####
## this file is tissue specific THSs obatined from "001ATAC_seq"

X = annotatePeak(paste0("D:/ATAC_peak/wgcna/wgcna_peak/",tissue,"_specific_peak_sort.bed"),tssRegion = c(-3000,3000),TxDb = poppy_TxDb)

perk_related_to_gene <- as.data.frame(as.GRanges(X))
perk_related_to_gene <- perk_related_to_gene[abs(perk_related_to_gene$distanceToTSS) <= 3000,]

wgcna_module <- read.csv(paste0("D:/ATAC_peak/wgcna/wgcna_result/",module,"_gene_name.csv"),stringsAsFactors = F,header = T)
colnames(wgcna_module) <- "gene"


atac_rna <- perk_related_to_gene[perk_related_to_gene$geneId %in% wgcna_module[,1],]
atac_rna <- atac_rna[,c(1,2,3,6,19,21)]

atac_rna[,2] <- atac_rna[,2]-1

density <- ggplot(perk_related_to_gene, aes(x = abs(distanceToTSS)))+ geom_density(color = "black", fill = "gray")

write.table(atac_rna,paste0("D:/ATAC_peak/wgcna/wgcna_peak_homer/",tissue,"_",module,".bed"),
            sep = "\t",row.names = F,col.names = F,quote = F)
```



## 1.2 Homer to find motif

```shell
cd /mnt/d/ATAC_peak/wgcna/wgcna_peak_homer/
awk '{print $4"\t"$1"\t"$2"\t"$3"\t+"}' stem_blue.bed >homer_peaks.tmp
conda activate homer
findMotifsGenome.pl ./homer_peaks.tmp /mnt/d/Papaver_somniferum.ref.fa stem_blue_motifDir -p 4
```



## 1.3 blast to identify motif related gene  name in opium poppy

```
Because many motif names like "athb6", we must known which gene in opium poppy is "athb6"
```

```R
rm(list=ls())
## get id from homer results
library(tidyverse)
tissue = "stem_blue"
homer_result <- read.table(paste0("D:/ATAC_peak/wgcna/wgcna_peak_homer/",tissue,"_motifDir/knownResults.txt"),sep = "\t",skip = 1)
colnames(homer_result) <- c("Motif Name","Consensus","P-value","Log P-value","q-value (Benjamini)","of Target Sequences with Motif","% of Target Sequences with Motif","of Background Sequences with Motif"	,"% of Background Sequences with Motif")
homer_result1 <- separate(data = homer_result, col = `Motif Name`, into = c("TF_name","data_set"), sep = "/")
homer_result1 <- homer_result1[homer_result1[,4]<=0.05&homer_result1[,6]<=0.05,]
homer_result1 <- separate(data = homer_result1, col = TF_name, into = "TF", sep = "\\(")
homer_result1 <- as.data.frame(homer_result1[,1])
write.table(t(homer_result1),paste0("D:/ATAC_peak/wgcna/wgcna_peak/",tissue,"/known_ATAC_tf_only_name_knownResults.txt"),sep = " ",col.names = F,row.names = F,quote = F)

```

```shell
cd /mnt/d/ATAC_peak/wgcna/wgcna_peak/JING_blue/
## "known_uniprot-ATAC_tf_only_name_knownResults.fasta" is a  protein fasta file.
## using "known_ATAC_tf_only_name_knownResults.txt" as name input to find protein sequences in uniprot database with arabidopsis species as ref.

blastp -db blastp -query known_uniprot-ATAC_tf_only_name_knownResults.fasta -evalue 1e-5 -num_threads 4 -outfmt 6 -qcov_hsp_perc 60.0 -best_hit_score_edge 0.05 -best_hit_overhang 0.25 -max_target_seqs 1  -out known_blast_result.txt

```



## 1.4 Get hub tfs

```R

library(tidyverse)
library(readxl)
library(XML)
library(bitops)
library(RCurl)
tissue <- "stem"
tissue_folder <- "stem_blue"
module_name <- "blue"

## this file is all protein blast result between arabidopsis and opium poppy

blastp_result_all <- read.table("D:/ATAC_peak/wgcna/wgcna_peak/blast_poppy_new_genome_vs_ara.txt",sep = "\t",stringsAsFactors = F)
## filter low quality results
blastp_result_all <- blastp_result_all[blastp_result_all[,3] >= 40,]
blastp_result_all <- blastp_result_all[blastp_result_all[,12] >= 100,]

## this file get from "1.3 blast to identify motif related gene  name in opium poppy"
blastp_result <- read.table(paste0("D:/ATAC_peak/wgcna/wgcna_peak/",tissue_folder,"/known_blast_result.txt"),sep = "\t",stringsAsFactors = F)
blastp_result <- merge(blastp_result,blastp_result_all,by.x="V2",by.y="V2")
blastp_result <- blastp_result[,c(2,13,3:12)]
colnames(blastp_result) <- colnames(blastp_result_all)
## filter low quality results
blastp_result <- blastp_result[blastp_result[,3] >= 40,]

rna_seq_tpm_normal <- read.csv("D:/rnaseq/tpm.csv",stringsAsFactors = F,header = T)
colnames(rna_seq_tpm_normal) <- c("gene","capsule","petal","stem","fineroot","leaf","taproot")

## filter low expressed genes
rna_seq_tpm_normal <- rna_seq_tpm_normal[rna_seq_tpm_normal[colnames(rna_seq_tpm_normal) == tissue] >=5,]
blastp_result <- blastp_result[blastp_result[,2] %in% rna_seq_tpm_normal[,1],]


tissue_homer <- blastp_result

tissue_homer <- tissue_homer[,c(1,2)]
tissue_homer <- separate(data = tissue_homer, col = V1, into = c("1","2","3"), sep = "\\|")
tissue_homer <- tissue_homer[,c(2,4)]
colnames(tissue_homer) <- c("uniprot_name","tf_poppy_name")

tf_information <- read_tsv(paste0("D:/ATAC_peak/wgcna/wgcna_peak/",tissue_folder,"/known_uniprot-ATAC_tf_only_name_knownResults.tab"),skip = 1,col_names = F)
tf_information <- tf_information[,c(1,2,5,7)]
colnames(tf_information) <- c("tf_name","protein_uniprot_name","description","Organism")

## this file obtained from itak
## we upload protein fasta file obtained from uniprot to itak in order to filter false postive results.
tf_information_itak <- read.csv(paste0("D:/ATAC_peak/wgcna/wgcna_peak/",tissue_folder,"/known_tf_classification.txt"),sep="\t",header = F,stringsAsFactors = F)
tf_information_itak <- separate(data = tf_information_itak, col = V1, into = c("1","2","3"), sep = "\\|")
tf_information_itak <- tf_information_itak[,c(2,6)]
tf_information <- merge(tf_information_itak,tf_information,by.x="2",by.y="protein_uniprot_name")
tf_information <- tf_information[,c(3,1,4:5)]
colnames(tf_information) <- c("tf_name","protein_uniprot_name","description","Organism")


tf_merge <- merge(tissue_homer,tf_information,by.x="uniprot_name",by.y = "protein_uniprot_name")

URL <- paste0("file:///D:/ATAC_peak/wgcna/wgcna_peak_homer/",tissue_folder,"_motifDir/knownResults.html")
temp <- getURL(URL)
web_tissue <- htmlParse(temp)
root <- xmlRoot(web_tissue)
length_motif <- as.data.frame(xmlSApply(root[["body"]][["table"]],xmlValue))
tissue_motif_matrix <- rbind()
for (i in 2:length(t(length_motif))) {
  tissue_motif_matrix <- rbind(tissue_motif_matrix,xmlToDataFrame(root[["body"]][["table"]][i]))
}
tissue_motif_matrix <- data.frame(lapply(tissue_motif_matrix,as.character),stringsAsFactors = F)
tissue_motif_matrix<- tissue_motif_matrix[,c(1,3)]
colnames(tissue_motif_matrix) <- c("rnak","Motif Name")


tissue_motif_matrix_true1 <- separate(data = tissue_motif_matrix, col = `Motif Name`, into = "TF", sep = "/")
tissue_motif_matrix_true1 <- separate(data = tissue_motif_matrix_true1, col = TF, into = "TF_name", sep = "\\(")

tf_poppy <- merge(tf_merge,tissue_motif_matrix_true1,by.x = "tf_name",by.y = "TF_name")
tf_poppy <- tf_poppy[tf_poppy[,1] != "MNT",]

tf_poppy[,7] <- paste0("/mnt/d/ATAC_peak/wgcna/wgcna_peak_homer/",tissue_folder,"_motifDir/knownResults/","known",tf_poppy[,6],".motif")

path <- as.data.frame(unique(tf_poppy[,7]))
path <- data.frame(lapply(path,as.character),stringsAsFactors = F)


len <- length(t(path))+1
path[len,1] <- "test"

write.csv(tf_poppy,paste0("D:/ATAC_peak/wgcna/wgcna_peak/",tissue_folder,"/",tissue_folder,"_tf_all_known.csv"),row.names = F,quote = F)
write.table(t(path),paste0("D:/ATAC_peak/wgcna/wgcna_peak/",tissue_folder,"/path_to_motif/",tissue_folder,"_tf_path_known.txt"),sep = " ",col.names = F,row.names = F,quote = F)


module_node <-read.csv(paste0("D:/ATAC_peak/wgcna/wgcna_result/",module_name,"_gene_name.csv"),stringsAsFactors = F)

## tfs also presented module identified in WGCNA were defined as hub-tfs in our study
tf_poppy <- tf_poppy[tf_poppy[,3] %in% module_node[,1],]
write.csv(tf_poppy,paste0("D:/ATAC_peak/wgcna/wgcna_peak/",tissue_folder,"/",tissue_folder,"_tf_hub_all_known.csv"),row.names = F,quote = F)

```

## 1.5 Get motif binding information

```shell
cd /mnt/d/ATAC_peak/wgcna/wgcna_peak/stem_blue/path_to_motif/
  
ls | while read id;do(cat $id |while read id2;do(cat $id2 >>/mnt/d/ATAC_peak/wgcna/wgcna_peak/stem_blue/$id.motifs);done);done
cd ..
conda activate homer
annotatePeaks.pl ./stem_blue.bed /mnt/d/Papaver_somniferum.ref.fa -m stem_blue_tf_path_known.txt.motifs > stem_blue_tf_path_known.txt.motifs.txt
```

```R
options(stringsAsFactors = FALSE)
tissue_folder <- "stem_blue"

## tf binf infor obtained from annotatePeaks scripts
motif_binding <- read.csv(paste0("D:/ATAC_peak/wgcna/wgcna_peak/",tissue_folder,"/",tissue_folder,"_tf_path_known.txt.motifs.txt"),sep="\t",stringsAsFactors = F)
a <- as.data.frame(table(motif_binding[,1]))
motif_binding <- as.data.frame(t(motif_binding[c(1,22:length(colnames(motif_binding)))]))


pos <- dim(motif_binding)[2]+1
pos_1 <- dim(motif_binding)[1]




motif_binding[,pos] <- rownames(motif_binding) 
motif_binding <- separate(data = motif_binding, col = paste0("V",pos), into = "TF_name", sep = "\\.")
motif_binding <- as.data.frame(t(motif_binding))


result_data <- rbind()


for (tf in 2:pos_1) {
  print(tf)
  result_data_1 <- rbind()
  for(target_gene in 1:(pos-1)){
    if(motif_binding[target_gene,tf] !=""){
      result_data_2 <- motif_binding[target_gene,1:2]
      result_data_2[1,2] <- as.character(motif_binding[pos,tf])
      colnames(result_data_2) <- c("peak","tf")
      result_data <- rbind(result_data,result_data_2)
    }
  }
}

peak <- read.csv(paste0("D:/ATAC_peak/wgcna/wgcna_peak/",tissue_folder,"/",tissue_folder,".bed"),sep = "\t",header = F)
peak <- peak[,c(4,5)]
result_data <- merge(result_data,peak,by.x="peak",by.y="V4")
colnames(result_data) <- c("peak","tf","target_gene")

poppy_tf <- read.csv(paste0("D:/ATAC_peak/wgcna/wgcna_peak/",tissue_folder,"/",tissue_folder,"_tf_all_known.csv"))
poppy_tf <- poppy_tf[,-c(5,6)]
result_data <- merge(result_data,poppy_tf,by.x="tf",by.y="tf_name")

write.csv(result_data,paste0("D:/ATAC_peak/wgcna/wgcna_peak/",tissue_folder,"/tf/","tf_target.csv"))


poppy_tf_hub <- read.csv(paste0("D:/ATAC_peak/wgcna/wgcna_peak/",tissue_folder,"/",tissue_folder,"_tf_hub_all_known.csv"))

result_data <- result_data[result_data[,5] %in% poppy_tf_hub[,3],]
result_data <- result_data[result_data[,1] %in% poppy_tf_hub[,1],]
write.csv(result_data,paste0("D:/ATAC_peak/wgcna/wgcna_peak/",tissue_folder,"/tf/","hub_tf_target.csv"))

```

## 1.6 Uniprot annotation

```R
library(UniprotR)
tissue <- "stem"
setwd("D:/ATAC_peak/wgcna/wgcna_peak/tf_anno/")
list <- read.csv(paste0(tissue,".csv"),stringsAsFactors = F)
colnames(list) <- c("tf","uniprot_name")
a <- GetProteinFunction(list[,2])
a <- a[,c(1,9)]
a[,1] <- rownames(a)
colnames(a) <- c("uniprot_name","general_function_description")
b <- GetProteinGOInfo(list[,2])
b[,1] <- rownames(b)
colnames(b) <- c("uniprot_name","GO_ALL","GO_BP","GO_MF","GO_CC")
tf_anno <- merge(a,b,by="uniprot_name")

list_merge <- merge(list,tf_anno,by="uniprot_name")
write.csv(list_merge,paste0(tissue,"_anno.csv"),na = " ")

```

